### PR TITLE
Adding 3 Advanced Snippets

### PR DESCRIPTION
**Pull request by fungktt**
I'm working on lcm.md, pullAtValue.md, pullAtIndex.md

### CODE REVIEW

File: snippets/pullAtIndex.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+title: pullAtIndex
+tags: array,advanced
+---
+Mutates the original array to filter out the values at the specified indexes. Returns the removed elements.
+
+Use `filter!()` to pull out the values that are not needed. Use `push!()` to re-populate a new array with only the values pulled.
+
+```jl
+function pullAtIndex(arr, pullArr)
```
Comment (abhishalya) [inline snippets/pullAtIndex.md]: It would be better to use snake_case as a stylistic convention.

Comment (logankilpatrick) [reply snippets/pullAtIndex.md]: technically, Julia convention is to use all lowercase unless it's hard to read.

**Comment by fungktt** [comment]
> Looks good from a content perspective. I would just update it with the correct function naming style.

@logankilpatrick Updated!

File: snippets/pull_at_index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+title: pull_at_index
+tags: array,advanced
+---
+Mutates the original array to filter out the values at the specified indexes. Returns the removed elements.
+
+Use `filter!()` to pull out the values that are not needed. Use `push!()` to re-populate a new array with only the values pulled.
+
+```jl
+function pull_at_index(arr, pullArr)
```
Comment (abhishalya) [inline snippets/pull_at_index.md]: snake_case :snake:

File: snippets/pull_at_index.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  removed=String[]
+  counter=0
+  for i in pullArr
```
Comment (abhishalya) [inline snippets/pull_at_index.md]: :snake:

File: snippets/pull_at_index.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    push!(removed, value)
+    counter=counter+1
+  end
+removed
+end
+```
+
+```jl
+myArray = ["a", "b", "c", "d"]
+pulled = pull_at_index(myArray, [1,3]) #myArray=["b", "d"]; pulled=["a", "c"]
```
Comment (abhishalya) [inline snippets/pull_at_index.md]: :snake: 

Also, space after `#` please

File: snippets/pull_at_value.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+title: pull_at_value
+tags: array,advanced
+---
+Mutates the original array to filter out the values specified. Returns the removed elements.
+
+Use `filter!()` to pull out the values that are not needed. Use `push!()` to re-populate a new array with only the values pulled.
+
+```jl
+function pull_at_value(arr, pullArr)
```
Comment (abhishalya) [inline snippets/pull_at_value.md]: :snake:

File: snippets/pull_at_value.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  removed=String[]
+  for i in pullArr
```
Comment (abhishalya) [inline snippets/pull_at_value.md]: :snake:

File: snippets/pull_at_value.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    filter!(x -> x != i, arr)
+    push!(removed, i)
+  end
+  removed
+end
+```
+
+```jl
+myArray = ["a", "b", "c", "d"]
+pulled = pull_at_value(myArray, ["b", "d"])  #myArray=["a", "c"]; pulled=["b", "d"]
```
Comment (abhishalya) [inline snippets/pull_at_value.md]: :snake:  & space

**Comment by abhishalya** [comment]
I'm pretty sure the maintainer will squash these into one, but you can do that too if you want. Try it :)

**Comment by Chalarangelo** [comment]
@PseudoCodeNerd Thanks for mentioning, I will delete it then.

**Pull request merged**